internal/resource/http.go

```diff
@@ -70,6 +80,103 @@ func (f *Fetcher) UpdateHttpTimeouts(timeouts types.Timeouts) {
 
 	f.client.transport.ResponseHeaderTimeout = time.Duration(responseHeader) * time.Second
 	f.client.client.Transport = f.client.transport
+
+	// Update CAs
+	if len(cas) == 0 {
+		return nil
+	}
+
+	pool, err := x509.SystemCertPool()
+	if err != nil {
+		f.Logger.Err("Unable to read system certificate pool: %s", err)
+		return err
+	}
+
+	for _, ca := range cas {
+		u, err := url.Parse(ca.Source)
+		if err != nil {
+			f.Logger.Err("Unable to parse CA URL: %s", err)
+			return err
+		}
+
+		cablob, err := f.getCABlob(u, ca)
+		if err != nil {
+			return err
+		}
+		block, _ := pem.Decode(cablob)
+		if block == nil {
+			f.Logger.Err("Unable to decode CA (%s)", u)
+			return ErrPEMDecodeFailed
+		}
+
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			f.Logger.Err("Unable to parse CA (%s): %s", u, err)
+			return err
+		}
+
+		f.Logger.Info("Adding %q to list of CAs", cert.Subject.CommonName)
+		pool.AddCert(cert)
+	}
+
+	f.client.transport.TLSClientConfig = &tls.Config{RootCAs: pool}
+	f.client.client.Transport = f.client.transport
+	return nil
+}
+
+func (f *Fetcher) getCABlob(u *url.URL, ca types.CaReference) ([]byte, error) {
+	if blob, ok := f.client.cas[ca]; ok {
+		return blob, nil
+	}
+	hasher, err := util.GetHasher(ca.Verification)
+	if err != nil {
+		f.Logger.Crit("Unable to get hasher: %s", err)
+		return nil, err
+	}
+
+	var expectedSum []byte
+	if hasher != nil {
+		// explicitly ignoring the error here because the config should already
+		// be validated by this point
+		_, expectedSumString, _ := ca.Verification.HashParts()
+		expectedSum, err = hex.DecodeString(expectedSumString)
+		if err != nil {
+			f.Logger.Crit("Error parsing verification string %q: %v", expectedSumString, err)
+			return nil, err
+		}
+	}
+
+	cablob, err := f.FetchToBuffer(*u, FetchOptions{
+		Hash:        hasher,
+		ExpectedSum: expectedSum,
+	})
+	if err != nil {
+		f.Logger.Err("Unable to fetch CA (%s): %s", u, err)
+		return nil, err
+	}
+	f.client.cas[ca] = cablob
+	return cablob, nil
+
+}
+
+// RewriteCAsWithDataUrls will modify the passed in slice of CA references to
+// contain the actual CA file via a dataurl in their source field.
+func (f *Fetcher) RewriteCAsWithDataUrls(cas []types.CaReference) error {
+	for i, ca := range cas {
+		u, err := url.Parse(ca.Source)
+		if err != nil {
+			f.Logger.Err("Unable to parse CA URL: %s", err)
+			return err
+		}
+
+		blob, err := f.getCABlob(u, ca)
```

Comment:
Thoughts on caching these instead of re-fetching them? We rewrite as data URLs to avoid refetching, but then refetch them to do so.

Reply:
All the CAs should be cached in `f.client.cas`, which `f.getCABlob` will check first, so no network traffic should happen here.

I can rewrite this to be unable to fail over to network traffic in the event of them not being in the cache, if you'd prefer.

Reply:
Oh whoops, missed that short return. This is good.